mayastor/src/grpc/mayastor/v1.rs

```diff
@@ -0,0 +1,196 @@
+
+use ::function_name::named;
+use std::panic::AssertUnwindSafe;
+
+#[derive(Debug)]
+pub struct MayastorSvc {
+    name: String,
+    interval: Duration,
+    rw_lock: tokio::sync::RwLock<Option<GrpcClientContext>>,
+}
+
+#[async_trait::async_trait]
+impl<F, T> Serializer<F, T> for MayastorSvc
+where
+    T: Send + 'static,
+    F: core::future::Future<Output = Result<T, Status>> + Send + 'static,
+{
+    async fn locked(&self, ctx: GrpcClientContext, f: F) -> Result<T, Status> {
+        let mut guard = self.rw_lock.write().await;
+
+        // Store context as a marker of to detect abnormal termination of the
+        // request. Even though AssertUnwindSafe() allows us to
+        // intercept asserts in underlying method strategies, such a
+        // situation can still happen when the high-level future that
+        // represents gRPC call at the highest level (i.e. the one created
+        // by gRPC server) gets cancelled (due to timeout or somehow else).
+        // This can't be properly intercepted by 'locked' function itself in the
+        // first place, so the state needs to be cleaned up properly
+        // upon subsequent gRPC calls.
+        if let Some(c) = guard.replace(ctx) {
+            warn!("{}: gRPC method timed out, args: {}", c.id, c.args);
+        }
+
+        let fut = AssertUnwindSafe(f).catch_unwind();
+        let r = fut.await;
+
+        // Request completed, remove the marker.
+        let ctx = guard.take().expect("gRPC context disappeared");
+
+        match r {
+            Ok(r) => r,
+            Err(_e) => {
+                warn!("{}: gRPC method panicked, args: {}", ctx.id, ctx.args);
+                Err(Status::cancelled(format!(
+                    "{}: gRPC method panicked",
+                    ctx.id
+                )))
+            }
+        }
+    }
+}
+
+impl TryFrom<CreatePoolRequest> for PoolArgs {
+    type Error = LvsError;
+    fn try_from(pool: CreatePoolRequest) -> Result<Self, Self::Error> {
+        if pool.disks.is_empty() {
+            Err(LvsError::Invalid {
+                source: Errno::EINVAL,
+                msg: "Missing devices".to_string(),
+            })
+        } else {
+            Ok(Self {
+                name: pool.name,
+                disks: pool.disks,
+            })
+        }
+    }
+}
+
+impl MayastorSvc {
+    pub fn new(interval: Duration) -> Self {
+        Self {
+            name: String::from("CSISvc"),
+            interval,
+            rw_lock: tokio::sync::RwLock::new(None),
+        }
+    }
+}
+
+impl From<Lvs> for Pool {
+    fn from(l: Lvs) -> Self {
+        Self {
+            name: l.name().into(),
+            disks: vec![l.base_bdev().bdev_uri().unwrap_or_else(|| "".into())],
+            state: PoolState::PoolOnline.into(),
+            capacity: l.capacity(),
+            used: l.used(),
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl mayastor_server::Mayastor for MayastorSvc {
+    #[named]
+    async fn create_pool(
```

Comment:
Dont we need to update the pool_cli also to point to the new functions?

Reply:
We should wait for all the code to migrate otherwise we may have two different clients in the cli